pkg/pillar/agentlog/agentlog.go

```diff
@@ -330,7 +329,7 @@ func InitWithDirText(agentName string, logdir string, curpart string) (*os.File,
 // Setup and return a logf, but don't redirect our log.*
 func InitChild(agentName string) (*os.File, error) {
 	logdir := GetCurrentLogdir()
-	return initImpl(agentName, logdir, false, false)
```

Comment:
Why do you need to change InitChild? You are not using it for dnsmasq.

Reply:
This is because of the way initImpl function has been changed for syslog support. Unless we pass true argument for text parameter, we do not even create the log file.

Reply:
OK, so there will be some followup for dhcpcd I assume

Reply:
Yes, I have already created a story for tracking. I was about to commit the change where I remove "-d" option that precents dnsmasq from backgrounding.
Will send another pull request.